### PR TITLE
GLTFLoader: Fix handling of GLTF files with optional KTX2 extensions

### PR DESCRIPTION
**Pull request by zeux**
If KTX2 loader wasn't registered, we would previously fail to load GLTF
files that used BasisU; this is a problem since files that don't require
BasisU support but specify it optionally would incorrectly fail.

The tradeoff here is that the error message that we previously emitted
for cases where the extension was required is now no more; we need some
mechanism for required extensions to actually be enforced to solve this.

### CODE REVIEW

**Comment by donmccurdy** [comment]
> The tradeoff here is that the error message that we previously emitted
for cases where the extension was required is now no more; we need some
mechanism for required extensions to actually be enforced to solve this.

I'm not sure we can make this tradeoff safely – it's going to be much more common for a user to forget to set up KTX2Loader than to load a model with fallback uncompressed textures. For now I think we should leave it as it is, but I agree that some mechanism for enforcing required extensions should be added when possible.

**Comment by zeux** [comment]
When you say "as is", do you mean to say that until we get a cleaner mechanism to distinguish between required and optional uses of the extension, it's reasonable to only support the required case since it's more prominent? Asking because this came out of #20508 that has a similar tradeoff, and I want to be consistent - if we should keep the error handling as it is today for now, I'd rather change that PR to match as well.

**Comment by donmccurdy** [comment]
If we need to make a (temporary) choice between (a) supporting fallback when a dependency is missing, and (b) showing useful errors when a dependency is missing, I'd take the latter. But if you see a clean way to add better error handling to the extension processing code and would be willing to add that, having both (a) and (b) sounds great!

**Comment by zeux** [comment]
@donmccurdy What do you think about this solution?

File: examples/js/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -696,8 +696,16 @@ THREE.GLTFLoader = ( function () {
 
 		if ( ! loader ) {
 
-			// Assumes that the extension is optional and that a fallback texture is present; ideally we should handle extensionsRequired better
-			return null;
+			if ( json.extensionsRequired && json.extensionsRequired.includes( this.name ) ) {
```
Comment (donmccurdy) [inline examples/js/loaders/GLTFLoader.js:699]: For the time being we are still using `.indexOf( str ) >= 0` to avoid needing transpilation for IE11.

**Review by donmccurdy — COMMENTED** [review]
Looks good, thanks! Just one change suggested.

**Review by donmccurdy — APPROVED** [review]
Thank you!

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**